File: COLLEGE/ASSIGNMENT8/python/Q.4.py
def find_odd_occurrence(arr):
    result = 0
    for num in arr:
        result ^= num
    return result

# Test
arr = [1, 2, 3, 2, 3, 1, 3]
odd_occurrence = find_odd_occurrence(arr)
print(f"Element occurring odd number of times: {odd_occurrence}")

File: COLLEGE/ASSIGNMENT8/c/Q.4.c
#include <stdio.h>

int findOddOccurrence(int arr[], int n) {
    int result = 0;
    for (int i = 0; i < n; i++) {
        result ^= arr[i];
    }
    return result;
}

int main() {
    int arr[] = {2, 3, 4, 3, 2, 4, 4};
    int n = sizeof(arr) / sizeof(arr[0]);
    int result = findOddOccurrence(arr, n);

    printf("The number occurring odd times is: %d\n", result);
    return 0;
}
